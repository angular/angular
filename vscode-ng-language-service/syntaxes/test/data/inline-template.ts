/* clang-format off */

/* Inline template recognition tests */
@Component({
  //// Property key/value test
  template: '<div></div>',

  //// String delimiter tests
  template: `<div></div>`,
  template: "<div></div>",
  template: '<div></div>',

  //// Parenthesization tests
  template: ( (( '<div></div>' )) ),

  //// Comments tests
  // template: '<div></div>'
  /*
   * template: '<div></div>'
   */
  /**
   * template: '<div></div>'
   */
})
export class TMComponent{}

/* Template syntax tests */
@Component({
  // Interpolation test
  template: '{{property}}',
})
export class TMComponent{}
