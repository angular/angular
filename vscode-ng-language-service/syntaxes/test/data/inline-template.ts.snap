>/* clang-format off */
#^^^^^^^^^^^^^^^^^^^^^^^ inline-template.ng
>
>/* Inline template recognition tests */
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-template.ng
>@Component({
#^^^^^^^^^^^^^ inline-template.ng
>  //// Property key/value test
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-template.ng
>  template: '<div></div>',
#^^ inline-template.ng
#  ^^^^^^^^ inline-template.ng meta.object-literal.key.ts
#          ^ inline-template.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#           ^ inline-template.ng
#            ^ inline-template.ng string
#             ^^^^^^^^^^^ inline-template.ng text.html.derivative
#                        ^ inline-template.ng string
#                         ^^ inline-template.ng
>
>  //// String delimiter tests
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-template.ng
>  template: `<div></div>`,
#^^ inline-template.ng
#  ^^^^^^^^ inline-template.ng meta.object-literal.key.ts
#          ^ inline-template.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#           ^ inline-template.ng
#            ^ inline-template.ng string
#             ^^^^^^^^^^^ inline-template.ng text.html.derivative
#                        ^ inline-template.ng string
#                         ^^ inline-template.ng
>  template: "<div></div>",
#^^ inline-template.ng
#  ^^^^^^^^ inline-template.ng meta.object-literal.key.ts
#          ^ inline-template.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#           ^ inline-template.ng
#            ^ inline-template.ng string
#             ^^^^^^^^^^^ inline-template.ng text.html.derivative
#                        ^ inline-template.ng string
#                         ^^ inline-template.ng
>  template: '<div></div>',
#^^ inline-template.ng
#  ^^^^^^^^ inline-template.ng meta.object-literal.key.ts
#          ^ inline-template.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#           ^ inline-template.ng
#            ^ inline-template.ng string
#             ^^^^^^^^^^^ inline-template.ng text.html.derivative
#                        ^ inline-template.ng string
#                         ^^ inline-template.ng
>
>  //// Parenthesization tests
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-template.ng
>  template: ( (( '<div></div>' )) ),
#^^ inline-template.ng
#  ^^^^^^^^ inline-template.ng meta.object-literal.key.ts
#          ^ inline-template.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#           ^ inline-template.ng
#            ^ inline-template.ng meta.brace.round.ts
#             ^ inline-template.ng
#              ^ inline-template.ng meta.brace.round.ts
#               ^ inline-template.ng meta.brace.round.ts
#                ^ inline-template.ng
#                 ^ inline-template.ng string
#                  ^^^^^^^^^^^ inline-template.ng text.html.derivative
#                             ^ inline-template.ng string
#                              ^ inline-template.ng
#                               ^ inline-template.ng meta.brace.round.ts
#                                ^ inline-template.ng meta.brace.round.ts
#                                 ^ inline-template.ng
#                                  ^ inline-template.ng meta.brace.round.ts
#                                   ^^ inline-template.ng
>
>  //// Comments tests
#^^^^^^^^^^^^^^^^^^^^^^ inline-template.ng
>  // template: '<div></div>'
#^^^^^ inline-template.ng
#     ^^^^^^^^ inline-template.ng meta.object-literal.key.ts
#             ^ inline-template.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#              ^ inline-template.ng
#               ^ inline-template.ng string
#                ^^^^^^^^^^^ inline-template.ng text.html.derivative
#                           ^ inline-template.ng string
>  /*
#^^^^^ inline-template.ng
>   * template: '<div></div>'
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-template.ng
>   */
#^^^^^^ inline-template.ng
>  /**
#^^^^^^ inline-template.ng
>   * template: '<div></div>'
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-template.ng
>   */
#^^^^^^ inline-template.ng
>})
#^^^ inline-template.ng
>export class TMComponent{}
#^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-template.ng
>
>/* Template syntax tests */
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-template.ng
>@Component({
#^^^^^^^^^^^^^ inline-template.ng
>  // Interpolation test
#^^^^^^^^^^^^^^^^^^^^^^^^ inline-template.ng
>  template: '{{property}}',
#^^ inline-template.ng
#  ^^^^^^^^ inline-template.ng meta.object-literal.key.ts
#          ^ inline-template.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#           ^ inline-template.ng
#            ^ inline-template.ng string
#             ^^ inline-template.ng text.html.derivative punctuation.definition.block.ts
#               ^^^^^^^^ inline-template.ng text.html.derivative expression.ng variable.other.readwrite.ts
#                       ^^ inline-template.ng text.html.derivative punctuation.definition.block.ts
#                         ^ inline-template.ng string
#                          ^^ inline-template.ng
>})
#^^^ inline-template.ng
>export class TMComponent{}
#^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-template.ng
>